### DISCUSSION TITLE
MDI / Layout view customization

### DISCUSSION THREAD

**Discussion by mouralabank** [CommunityToolkit/Maui#1061] [New Feature Discussions]
Can we have multiple document interface support (MDI) or something similar in MAUI? For Desktop applications is something very important and brings several benefits, flexibility and facilities.

Generally desktop platforms have many tools and every user has a way to assemble their workspace. With MDI the user can open multiple tools, move, minimize, maximize, resize and configure their work area as desired and is more useful in each situation.

**Comment by mouralabank** [comment]
Other benefits of MDI for desktop applications:

- Improved organization: MDI allows users to organize multiple documents or windows within a single parent window, making it easier to manage and navigate between them.
- More efficient use of screen space: By using MDI, desktop applications can maximize screen real estate by consolidating multiple windows into a single parent window.
- Better task management: Users can switch between open documents or windows more easily, as they are all contained within a single parent window.
- Reduced clutter: MDI can help reduce desktop clutter by consolidating multiple windows into a single parent window, making it easier to find the document or window you need.
- Consistent user experience: By using a common interface for multiple documents or windows, MDI can provide a consistent user experience across different parts of the application.



MDI can help improve productivity and user experience in desktop applications by providing a more efficient and organized way to manage multiple documents or windows

**Comment by bijington** [comment]
How would this work on mobile platforms?

**Reply by mouralabank** [reply]
MDI (Multiple Document Interface) is primarily a desktop user interface concept, although some mobile applications and operating systems have implemented MDI-like features to a certain extent.

In desktop environments, MDI allows users to work with multiple documents or windows within a single parent window, providing better organization, multitasking, screen real estate utilization, and collaboration features. This feature is commonly seen in office productivity software, such as Microsoft Word and Excel.

However, due to the smaller screen size and touch-based interaction of mobile devices, implementing MDI on mobile platforms can be more challenging. As a result, many mobile applications use alternative user interface designs to provide similar functionality, such as tabs, split-screen, and navigation drawers. These elements provide a similar experience of allowing users to work with multiple documents or windows within a single parent view, but optimized for the smaller screens and touch-based interaction of mobile devices.

**Reply by mouralabank** [reply]
Here are some ways that MDI could work on mobile platforms:

-Tabs: Many mobile applications use tabs to display multiple documents or windows within a single parent window. This approach can work well on mobile devices because it saves screen space and allows users to switch between documents quickly.

-Split-screen: Some mobile devices, such as tablets, support split-screen functionality, which can be used to display multiple documents or applications side by side. This approach can be particularly useful for users who need to reference multiple documents or applications simultaneously.

-Navigation drawer: A navigation drawer is a common user interface element in mobile applications that allows users to access different sections of the app. This element can be used to switch between different documents or windows within an MDI interface.

-Gestures: Mobile devices support a variety of touch gestures, such as swiping and pinching, that can be used to navigate between different documents or windows within an MDI interface.

**Reply by bijington** [reply]
Do you have any concrete implementation details that you would like to propose in order to progress this proposal or is this more just an idea?

**Reply by mouralabank** [reply]
I have some details for the implementation and structure of the proposal but I believe there is room for more ideas and suggestions

**Comment by mouralabank** [comment]
MDI is widely used in financial/trading market applications. Where the system has several features and functionalities and each user has a preference of which and how to use, organization, size of each tool, being able to move and change places, minimize and maximize a resource…

Here are a few examples of desktop applications used in the financial market and trading that use MDI:

Bloomberg Terminal: The Bloomberg Terminal is a software platform used by traders and investors for financial data analysis and trading. It uses MDI to allow users to work with multiple data screens and windows within a single parent window.

MetaTrader 4: MetaTrader 4 is a popular trading platform used by forex traders. It uses MDI to allow traders to work with multiple charts, indicators, and windows within a single parent window.

eSignal: eSignal is a market data and charting software used by traders for real-time market analysis and trading. It uses MDI to allow users to work with multiple charts and windows within a single parent window.

NinjaTrader: NinjaTrader is a trading platform used by futures and forex traders. It uses MDI to allow traders to work with multiple charts, indicators, and windows within a single parent window.
Interactive Brokers TWS: Interactive Brokers TWS (Trader Workstation) is a trading platform used by professional traders for executing trades and managing portfolios. It uses MDI to allow users to work with multiple order entry screens and windows within a single parent window.

**Reply by mouralabank** [reply]
Here are a few more examples of desktop applications that use MDI:

Microsoft Access: Microsoft Access is a database management system that uses MDI to allow users to work with multiple tables, queries, forms, and reports within a single parent window.

Notepad++: Notepad++ is a free and open-source code editor for Windows that uses MDI to allow users to work with multiple code files within a single parent window.

Adobe Acrobat: Adobe Acrobat is a PDF viewer and editor that uses MDI to allow users to work with multiple PDF files and windows within a single parent window.

Microsoft Project: Microsoft Project is a project management software that uses MDI to allow users to work with multiple project plans and windows within a single parent window.

Inkscape: Inkscape is a free and open-source vector graphics editor that uses MDI to allow users to work with multiple documents and windows within a single parent window.

Blender: Blender is a 3D creation software that uses MDI to allow users to work with multiple 3D models, scenes, and windows within a single parent window.

Microsoft Excel: Microsoft Excel is a spreadsheet software that uses MDI to allow users to work with multiple spreadsheets and windows within a single parent window.

Adobe Illustrator: Adobe Illustrator is a vector graphics editor that uses MDI to allow users to work with multiple documents and windows within a single parent window.

Microsoft Publisher: Microsoft Publisher is a desktop publishing software that uses MDI to allow users to work with multiple publications and windows within a single parent window.
Sublime Text: Sublime Text is a popular code editor that uses MDI to allow users to work with multiple code files and windows within a single parent window.

Corel PaintShop Pro: Corel PaintShop Pro is an image editing software that uses MDI to allow users to work with multiple images and windows within a single parent window.

Eclipse: Eclipse is an open-source IDE used for developing Java applications that uses MDI to allow users to work with multiple code files and windows within a single parent window.

**Comment by danielancines** [comment]
This implementation will be great for Desktop applications, I hope that arrive soon.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
